### PR TITLE
Fix: Miner heuristics for handling large transactions

### PR DESCRIPTION
**Pull request by kantai**
This fixes some issues with transaction inclusion for miners -- this is the same patch as #2438, but this is opened against `develop`, so that we can do a release of `develop` alongside this.

### CODE REVIEW

File: src/core/mod.rs [hunk 1]
```diff
@@ -48,6 +48,8 @@ pub const NETWORK_P2P_PORT: u16 = 6265;
 // its current block-commit in a sortition
 pub const MINING_COMMITMENT_WINDOW: u8 = 6;
 
+pub const TX_BLOCK_LIMIT_PROPORTION_HEURISTIC: u64 = 20;
```
Comment (lgalabru) [inline src/core/mod.rs:51]: Can we add a comment explaining how this constant is being used and why we came up with 20?

**Review by lgalabru — APPROVED** [review]
Looks good to me, thank you @kantai!
Just a tiny documentation request :)

File: src/core/mod.rs (hunk 1 above)
Comment (kantai) [reply src/core/mod.rs:57]: Added in 973a6aad8b0896e3aa9abc6dd4aaf2ce5fc17a4f

**Review by jcnelson — APPROVED** [review]
This LGTM.  But before merging to `master`, I think we should try this out on a testnet miner.

**Pull request merged**